torchtune/utils/memory.py

```diff
@@ -29,6 +31,12 @@ def set_activation_checkpointing(
     apply_activation_checkpointing(model, auto_wrap_policy=wrap_policy, **kwargs)
 
 
+def cleanup_before_training() -> None:
+    gc.collect()
+    torch.cuda.empty_cache()
+    torch.cuda.reset_peak_memory_stats()
```

Comment:
This makes sense, though do note now that during training, we might not get an accurate picture from the training logs on what is our true peak memory  across the entire job. I think this is fine, because if peak memory reserved occurs during model init, we'll still log that at the beginning, and then proceed to log memory peaks observed during training (that might be less).